packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -188,45 +239,85 @@ export const prepareResolverKit = (
         },
       },
       reporter: {
+        upsertPendingTx(
+          txId: TxId,
+          txMeta: TxMeta,
+          status: TxStatus,
+          rejectionReason?: string,
+        ) {
+          /**
+           * FIXME: IBC_FROM_REMOTE transactions are not yet supported by the
+           * resolver. We emulate the old behaviour in this case; just resolve
+           * the transaction as soon as we've identified it.
+           */
+          {
+            if (txMeta?.type === 'IBC_FROM_REMOTE') {
+              const { vowKit } = this.state.transactionRegistry.get(txId);
+              vowKit.resolver.resolve(
+                // @ts-expect-error cast
+                'IBC_FROM_REMOTE not implemented by resolver; auto-resolving',
+              );
+              status = TxStatus.SUCCESS;
+            }
+          }
+
+          const {
+            type,
+            destinationAddress,
+            amountValue: amount,
+            expectedAddr,
+            ...rest
+          } = txMeta;
+
+          const value: PublishedTx = {
+            type,
+            ...rest,
+            ...(destinationAddress ? { destinationAddress } : {}),
+            ...(txsWithAmounts.includes(type) ? { amount } : {}),
+            ...(type === TxType.MAKE_ACCOUNT ? { expectedAddr } : {}),
+            ...(rejectionReason ? { rejectionReason } : {}),
+            status,
+          };
+
+          const node = E(pendingTxsNode).makeChildNode(txId);
+          writeToNode(node, value);
+        },
+
         insertPendingTransaction(
           txId: TxId,
           destinationAddress: AccountId,
           type: TxType,
           amount?: NatValue,
           expectedAddr?: `0x${string}`,
         ) {
-          const value: PublishedTx = {
+          const txMeta: TxMeta = {
             type,
             destinationAddress,
             status: TxStatus.PENDING,
             ...(txsWithAmounts.includes(type) ? { amount } : {}),
             ...(type === TxType.MAKE_ACCOUNT ? { expectedAddr } : {}),
           };
-          const node = E(pendingTxsNode).makeChildNode(txId);
-          writeToNode(node, value);
+          return this.facets.reporter.upsertPendingTx(
+            txId,
+            txMeta,
+            TxStatus.PENDING,
+          );
         },
 
         completePendingTransaction(
           txId: TxId,
-          status: Exclude<TxStatus, 'pending'> = TxStatus.SUCCESS,
+          status: Exclude<TxStatus, 'PENDING' | 'SETUP'> = TxStatus.SUCCESS,
```

Comment:
is this casing a breaking change?

Reply:
It was my misunderstanding of the `TxStatus` enum.  Fixed.